Nerf eating fairies

AlexNipal opened the pull request:
Scale the Vore experience gain to scale with the weigth of the victim. This makes Fairies give less experience than a Banditlord.
You should gain experience a bit faster if you keep yourself full (unless you were abusing fairies)
The experience is scaled differently for each organ so you can gain exp at the same pace (based on early organs, will need tweaking).

chikincheese commented:
This is *almost* good enough - it should also consider the *initial* weight of your victim, imo, to make consuming larger enemies a better option.
(I also want to put back my "only stretching when not digesting" function, and have that also be a perk.)

Ugre101 commented:
Hmm think I will do a similar method but use stomach capacity vs max capacity as a percentage modifier. 
Stomach fullness * stomach current weight = exp gain

So that players are rewarded for keeping a full stomach rather the the amount of prey. 
Moving the exp gain outside the for loop.

chikincheese commented:
Sounds good. The other thing (once again, going back to stretching - not surprising at all) is that your capacity should only increase when you're at or near your limit - maybe >=90% of capacity?

(Later on, I think we should tie max capacity to sizes - either restrict max capacity by each part's size, or increase its size based on capacity. Both only work for vore, though, which will need to be considered later on for balance.)

Ugre101 commented:
X > 90 is a bit unforgiving that would need support for overextending over 100%.

Players shouldn't have to do math on what to eat.

chikincheese commented:
It's not a math worry; if you're not full (which I arbitrarily set at 90%), then keep consuming!
Can't fit anything else in, but nothing's small enough? Digest until you've got space.
As long as ([bodypart].capacity > [bodypart].maxcapacity * q), where q is how full you need to be, you increase capacity.

I could make different 'fullness levels' - 0% is empty, under 20% for lacking, under 40% for satisfied, under 60% for half-filled, under 80% for stuffed, else bulging.

AlexNipal commented:
Checking the initial weight would not help: you can eat a halfling, digest it until it's almost dead at 1kg, regurgitate it and re-insert it in an other hole where you stop digesting it.

Using the maximum capacity and different thresholds is a great idea, I'll try to implement it.

chikincheese commented:
Well, you'd check (and set) their "initialWeight" whenever they're consumed - if you'd get 0.0001 xp/update normally, change it to 0.00001 * initialWeight.
Offers an incentive to eat larger enemies (for obvious reasons), and *partially* counteracts any benefits from fairies.
Another option [read: idea] is that xp gain is tied to height.

AlexNipal commented:
That could still be abused somewhat: if you stuff yourself then almost digest the victims and stop before they expire, you can stuff yourself once more and get... up to twice the experience, this doesn't seem so bad actually. I'll give it a shot.

Using height is a neat idea, but I think we would need to refactor the whole system to use height (volume) instead of weight.

chikincheese commented:
Well, as I suggested, the calculation for initialheight and initialweight would be called within "Vore.js" to stop that from being abused.
In the meanwhile, I need to see how I broke my own system. (Going into negative values, somehow, for vore points.)

AlexNipal commented:
I implemented the experience using maximum capacity and startWeigth (which was already tracked). I also added a parameter so that things can accelerate Vore.

Now I have a merge conflict. I am fixing the conflicts but I will finish tomorrow.

AlexNipal commented:
This branch is now up to date with Master and ready to be reviewed again.
